bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/validation/BigtableSyncTableJob.java

```diff
@@ -0,0 +1,483 @@
+    System.err.println(" sourcetable      Name of the source table to sync from");
+    System.err.println(" targettable      Name of the target table to sync to");
+    System.err.println();
+    System.err.println("Examples:");
+    System.err.println(" For SyncTable validation of tableA from a remote source cluster");
+    System.err.println(" to a local target cluster:");
+    System.err.println(
+        " $ bin/hbase "
+            + "com.google.cloud.bigtable.mapreduce.validation.SyncTable --targetbigtableproject=project123"
+            + " --targetbigtableinstance=instance123 --sourcezkcluster=zk1.example.com,zk2.example.com,zk3.example.com:2181:/hbase"
+            + " gs://bucket/hashes/tableA tableA tableA");
+  }
+
+  /**
+   * Set up job configuration for Bigtable as Target
+   *
+   * @param otherArgs
+   * @return
+   * @throws IOException
+   */
+  private Job setupJobWithBigtable(String[] otherArgs) throws IOException {
+    // set up default job configurations
+    Job job = super.createSubmittableJob(otherArgs);
+    Configuration jobConf = job.getConfiguration();
+
+    // set conf for job startup as InputFormat is initialized on target
+    String targetZkCluster = BigtableSyncTableAccessor.getTargetZkCluster(this);
+    if (targetZkCluster != null) {
+      ZKConfig.ZKClusterKey zkClusterKey = ZKConfig.transformClusterKey(targetZkCluster);
+      jobConf.set(HConstants.ZOOKEEPER_QUORUM, zkClusterKey.getQuorumString());
+      jobConf.setInt(HConstants.ZOOKEEPER_CLIENT_PORT, zkClusterKey.getClientPort());
+      jobConf.set(HConstants.ZOOKEEPER_ZNODE_PARENT, zkClusterKey.getZnodeParent());
+    }
+
+    // enable dry run as default
+    String DRY_RUN_CONF_KEY = BigtableSyncTableAccessor.getConfDryRunKey();
+    jobConf.setBoolean(DRY_RUN_CONF_KEY, jobConf.getBoolean(DRY_RUN_CONF_KEY, true));
+
+    // Set up bigtable configurations for job. Note that the job conf is shared for source and
+    // target databases and additional configuration in mapper initializes proper
+    // connections.
+    String sourceZkCluster = BigtableSyncTableAccessor.getSourceZkCluster(this);
+    if (sourceBigtableProjectId != null && sourceBigtableInstance != null) {
+      if (sourceZkCluster != null)
+        LOG.warn(
+            "sourceZkCluster config("
+                + sourceZkCluster
+                + ") overriden with sourceBigtableProjectId("
+                + sourceBigtableProjectId
+                + "), sourceBigtableInstance("
+                + sourceBigtableInstance
+                + ")");
+
+      jobConf.set(SOURCE_BT_PROJECTID_CONF_KEY, sourceBigtableProjectId);
+      jobConf.set(SOURCE_BT_INSTANCE_CONF_KEY, sourceBigtableInstance);
+
+      if (sourceBigtableAppProfile != null)
+        jobConf.set(SOURCE_BT_APP_PROFILE_CONF_KEY, sourceBigtableAppProfile);
+    }
+    if (targetBigtableProjectId != null && targetBigtableInstance != null) {
+      if (targetZkCluster != null)
+        LOG.warn(
```

Comment:
I am not sure if this is before or after the "set fake connections for initializations" step. But if the customer supplies both targetZkCluster and targetBtProject, we should fail the job.

Reply:
with the existing behavior, if both are set, targetZkCluster and targetBtProject, then we default to Bigtable. The same is true for sourceZkCluster and sourceBigtableProjectId. If both are set, we default to sourceBigtableProjectId for establishing a connection. I agree that we should fail if both are explicitly set as that means the user isn't properly configuring the job. will update this.